vector-im/element-android
# [Location sharing] - Message for live sharing in timeline (PSF-884)

## Type of change


## Content


Handling of the UI for the message associated to a live location share in the timeline. The message stays at the same position (at the timestamp of the start of the live). The message has 3 states:
- Loading (live has been start but no location has been received yet)
- Running (showing the last location of the user who is sharing)
- Ended (the live has been stopped by the user or is timed out)

The UI will also change depending if the user is the creator of the live or not.

Known issues that will be fixed later:
- Long waiting after pressing stop: a loader will be shown after pressing "Stop"
- No refresh of the message if the user is viewing the timeline and the live goes from running state to timed out
- Blinking of the map when location is updated

## Motivation and context


Closes #5689 

## Screenshots / GIFs



_Note: in the running state on emitter side, the live locatiion sharing top bar is not visible is some of the captures due to missing handling when restarting the app (like when toggling dark mode)_ 
|Running state (Emitter/default layout) |Running state (Emitter/bubble layout)|
|-|-|
|<img src="https://user-images.githubusercontent.com/46314705/167811802-5dad6da1-1a2e-403e-8930-5d431c92460f.png" width=150/><img src="https://user-images.githubusercontent.com/46314705/167811809-4af97a5f-c48c-4ff7-968b-413d413a809e.png" width=150/>|<img src="https://user-images.githubusercontent.com/46314705/167812157-21b3f5ee-a46d-49a7-abc9-bbcfda875819.png" width=150/><img src="https://user-images.githubusercontent.com/46314705/167812162-88bdd747-23a4-4e7f-9815-377eb5678c4c.png" width=150/>|

|Running state (Watcher/default layout) |Running state (Watcher/bubble layout)|
|-|-|
|<img src="https://user-images.githubusercontent.com/46314705/167813364-0ebda1d5-29e3-4d91-b87c-f4852167342c.png" width=150/><img src="https://user-images.githubusercontent.com/46314705/167813371-105b4d36-0d7c-4ad4-b8ce-5ca6f947983e.png" width=150/>|<img src="https://user-images.githubusercontent.com/46314705/167813374-82720c1c-d508-404d-95dc-6f3a8dbd6402.png" width=150/><img src="https://user-images.githubusercontent.com/46314705/167813380-f2c199a1-90fd-49c1-ab7e-58c3e636a9f2.png" width=150/>|

|Ended state (default layout) |Ended state (bubble layout)|
|-|-|
|<img src="https://user-images.githubusercontent.com/46314705/167812375-e515c56d-3795-4dba-8e4a-1f892222006a.png" width=150/><img src="https://user-images.githubusercontent.com/46314705/167812379-2745226f-bec5-4226-83cc-b7bbfa36396c.png" width=150/>|<img src="https://user-images.githubusercontent.com/46314705/167812384-8182746a-1982-44bd-9a75-5220b31dc1a2.png" width=150/><img src="https://user-images.githubusercontent.com/46314705/167812389-fd80f800-3b1a-4381-8ee9-864f691b1c11.png" width=150/>|

## Tests



- Ensure to enable location sharing in Settings -> Preferences
- Ensure to enable developer mode in Settings -> Advanced settings
- Ensure to enable the live location feature in the developer setting
- Go to a room where you can send state event
- Press the plus button on the new message layout
- Press the location icon
- Choose the "Live location" option
- Check the message displays "Loading" state
- Check the message displays the correct last location
- Check the location is updated when moving (can be emulated by a fake route on emulator)
- Press the stop button
- Check the message displays the "Ended" state

## Tested devices

- OS version(s): Android 11

## Checklist
